### DISCUSSION TITLE
How to execute action without using namespace

### DISCUSSION THREAD

**Discussion by komdil** [OData/AspNetCoreOData#566] [Q&A]
Assemblies affected
Microsoft.AspNetCore.OData 7.5.12

Describe the bug
I didn't understand the purpose of namespace in ODataConventionModelBuilder. By default it will be set to Default. And for request we need to use namespace to execute action (like .../Default.ActionName) or casting properties. For me it looks stange, and I would like to not have namespace on executing actions.
So, my questions are:

Can we run action without namespace? Like .../ActionName
Can we have multiple namespaces? For example, I would like to work my actions on :

.../Default.ActionName
.../Default2.ActionName
Reproduce steps
Just create any action and run it using Default namespace

Data Model
Just create entity with action and map it

EDM (CSDL) Model
Please share your Edm model, for example, CSDL file.
You can send $metadata to get a CSDL XML content.

Expected behavior
I would like do not have a namespace

**Comment by xuzhg** [comment]
Can we run an action without namespace? Like .../ActionName
==>
   Yes, You can. By conventional routing, we build the endpoint for action/function with and without the namespace by default.
   So, by default, you have at least two endpoints for one endpoint of action/function.
   Meanwhile, we also provide RouteOptions to disable namespace or non-namespace endpoints.

Can we have multiple namespaces? For example, I would like to work my actions on :

==> Absolutely, yes. The action supports overload in one schema (need follow-up action overload rule). In different schema (namespace), You can have the same action name defined.
